Application/src/Model/diagramObject.java

```diff
@@ -0,0 +1,218 @@
+package Model;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+
+/**
+ * @author Rashad Kamsheh & Isabelle Törnqvist
+ * @since 2017-10-16
+ */
+
+public class diagramObject {
+
+
+    //handle diagram's meta
+    private Meta meta;
+    //handle the Type of the diagram
+    private String type;
+    //handle the Processes of the diagram
+    private List<Processes> processes = null;
+    //handle the messages of the diagram
+    private Diagram diagram;
+
+
+    public Meta getMeta() {
+        return meta;
+    }
+
+    public void setMeta(Meta meta) {
+        this.meta = meta;
+    }
+
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+
+    public void setProcesses(List<Processes> processes) {
+        this.processes = processes;
+    }
+
+    public List<Processes> getProcesses() {
+        return processes;
+    }
+
+
+    public Diagram getDiagram() {
+        return diagram;
+    }
+
+    public void setDiagram(Diagram diagram) {
+        this.diagram = diagram;
+    }
+
+
+
+    public class Meta {
```

Comment:
Would suggest to seperate classes into their own files, to follow the MVC structure

Reply:
I agree. No repeat of asteroids please.